pallets/did/src/lib.rs

```diff
@@ -423,6 +450,16 @@ pub mod pallet {
 
 			let did_identifier = details.did.clone();
 
+			// TODO: how to ensure reserve AND withdraw!?
```

Comment:
S: Remove the TODO. 

Your solution of checking the reservability of `Deposit + Fee` seems fine to me.
```suggestion
```